### PR TITLE
[16.0][MIG] sale_timesheet_rounded: Migration to 16.0

### PR DESCRIPTION
**Pull request by hoangphuc1410**
In the version 16.0, we have function _timesheet_preprocess is always update 'project_id' in model 'account.analytic.line'. So we don't have case no project_id in function 'test_analytic_line_read_override'. So I was delete line with project_id = False.
I remove WARNING in README because `force_compute` was removed from version 13.0 .

### CODE REVIEW

**Comment by StephaneMangin** [comment]
Please take in account this PR that should be needed... or not. Thanks

https://github.com/OCA/timesheet/pull/590

**Comment by sonhd91** [comment]
> Please take in account this PR that should be needed... or not. Thanks
> 
> #590

It's still needed, branch update included these commit, please check again

**Comment by leemannd** [comment]
Hello @hoangphuc1410 can you check the pre-commit hook?

**Comment by nilshamerlinck** [comment]
@sonhd91

**Comment by nilshamerlinck** [comment]
@sonhd91

**Comment by leemannd** [comment]
Following OCA wiki, you should squash together administrative commits https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0

**Comment by sonhd91** [comment]
> Following OCA wiki, you should squash together administrative commits https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0

Updated

**Review by leemannd — APPROVED** [review]

File: sale_timesheet_rounded/tests/test_rounded.py
Comment (leemannd) [inline sale_timesheet_rounded/tests/test_rounded.py]: Any reason to remove this test?

**Review by geomer198 — APPROVED** [review]

**Review by alexey-pelykh — APPROVED** [review]
LGTM

**Comment by ivs-cetmix** [comment]
Hey @OCA/project-service-maintainers any chances to have this merged?

**Comment by nilshamerlinck** [comment]
Hello @alexey-pelykh I just had a quick look at the failed checks and it seems to me they were just impacted by a transient issue, would you mind issuing the merge command again?

**Comment by nilshamerlinck** [comment]
omg! @alexey-pelykh my bad, I misread the situation completely :/
- merge command had been aborted due to failed checks and said failed checks did indeed fail because of transient networking issues
- BUT the real problem here is that the workflow is stuck in an infinite loop of `[UPD] Update hr_timesheet_sheet.pot`
- this has to do with the fact that `sale_timesheet_rounded` is in `rebel_module_groups`, and makepot runs in both odoo and OCB codebase, which it should not (as they are a bit different)
- I will push a fix to this PR but meanwhile it would be great if you could try to cancel all queued workflows for branch [16.0-ocabot-merge-pr-604-by-alexey-pelykh-bump-nobump](https://github.com/OCA/timesheet/commits/16.0-ocabot-merge-pr-604-by-alexey-pelykh-bump-nobump) in [https://github.com/OCA/timesheet/actions](https://github.com/OCA/timesheet/actions) if you have enough access rights for that

**Comment by nilshamerlinck** [comment]
Hi @dreispt @alexey-pelykh 

- I pushed a [fix](https://github.com/OCA/timesheet/pull/604/commits/29df840aa2dd8a6aae542b55f52c892ed73c8805) in this PR
- so you can now issue the merge command again :)

Details:
- In OCA/OCB codebase, `iap` module is `auto_install=False` since [this PR](https://github.com/OCA/OCB/pull/1143)
- As a consequence, `iap_mail` and `sms` are not auto-installed as well in the OCB build
- So field [`mail.thread:message_has_sms_error`](https://github.com/odoo/odoo/blob/1e76dda6da84bae8abfd0d0a41d9881a371f6d8a/addons/sms/models/mail_thread.py#L16) is not in db
- In `hr_timesheet_sheet` module, `hr_timesheet.sheet` [inherits from `mail.thread`](https://github.com/OCA/timesheet/blob/8d0abeb75ae6180021dab47cd7da09c2fc906fe2/hr_timesheet_sheet/models/hr_timesheet_sheet.py#L25)
- So pot file will differ when generated on OCB flavor
- When `sale_timesheet_rounded` [was marked as a rebel module](https://github.com/OCA/timesheet/pull/604/commits/36a6062a662d9f5f41f35a27243ff51bea181a5d), a trailing `makepot: "true"` was incorrectly kept so `makepot` was running on both odoo flavors
- `makepot` can run multiple times but it should always be on the same flavor

**Comment by alexey-pelykh** [comment]
@nilshamerlinck then it sounds that the fix should be squashed to the commit that made the module rebel

**Comment by nilshamerlinck** [comment]
Hello @alexey-pelykh the commit that made the module rebel was coming from [a PR targeting 15.0](https://github.com/OCA/timesheet/pull/590) but ok, no problem, I'll add a comment there to make sure it's updated as well there before it's merged.

Updated as per your request, can be merged :-)

**Pull request merged**
